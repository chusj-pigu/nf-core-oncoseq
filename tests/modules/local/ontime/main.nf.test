nextflow_process {

    name "Test Process ONTIME_RANGE_FILTER"
    script "modules/local/ontime/main.nf"
    process "ONTIME_RANGE_FILTER"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = tuple(
                    [id: "test_sample"],
                    file("${projectDir}/test_data/alignments/sample1_mito_align.bam"),
                    file("${projectDir}/test_data/alignments/sample1_mito_align.bam.bai"),
                    "0",
                    "24"
                )
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
