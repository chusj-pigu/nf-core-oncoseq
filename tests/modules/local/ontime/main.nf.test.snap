{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "sample1_mito_align.bam:md5,e9b55d9c712b30a595ebe7b25cfa3a4e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b75cf2241f8f497bc0973b765b258de2"
                ],
                "bam": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "sample1_mito_align.bam:md5,e9b55d9c712b30a595ebe7b25cfa3a4e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b75cf2241f8f497bc0973b765b258de2"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-06-03T19:41:38.720242846"
    }
}